src/cassio/table/utils.py

```diff
@@ -19,3 +19,76 @@ def error_handler(exc: BaseException) -> None:
 
     response_future.add_callbacks(success_handler, error_handler)
     return await asyncio_future
+
+
+def handle_multicolumn_unpacking(
+    args_dict: Dict[str, Any],
+    key_name: str,
+    unpacked_keys: List[str],
+) -> Dict[str, Any]:
+    """
+    Given a dictionary of "args", handle if necessary the replacement of one of its items
+    with corresponding split-tuple version if the type structure requires it.
+    So if unpacked_keys is == [key_name], do nothing.
+    If the key_name is None, remove it from the mapping altogether.
+    Returns the modified dictionary.
+
+    Example:
+        args_dict = {"k": (1, 20), "x": "..."}
+        key_name = "k"
+        unpacked_keys = ["k_0", "k_1"]
+    results in
+        {"k_0": 1, "k_1": 20, "x": "..."}
+
+    Example:
+        args_dict = {"k": "k_val", "x": "..."}
+        key_name = "k"
+        unpacked_keys = ["k"]
+    results in (unchanged)
+        {"k": "k_val", "x": "..."}
+
+    Example:
+        args_dict = {"x": "..."}
+        args_dict = {"k": None, "x": "..."}
+    both result in
+        {"x": "..."}
+    """
+    _unp_keys = list(unpacked_keys)
```

Comment:
i don't understand why you are creating this shallow copy of the existing list...

Reply:
Possibly in the future we might pass a generator or some other Iterable to the method, so it's better to materialize it as a list for the rest of the function to work safely.
You gave me the very good suggestion of broadening the type hint of the parameter, thank you!